pkg/codegen/java/utilities.go

```diff
@@ -11,23 +11,94 @@ import (
 	"github.com/pulumi/pulumi/pkg/v3/codegen"
 )
 
-func formatBlockComment(comment string, indent string) string {
+// formatForeignBlockComment formats comment as a java block comment for use in the
+// generated SDK. formatForeignBlockComment is responsible for ensuring that comment is
+// safe to embed in the generated SDK. That means that comment cannot contain any codes
+// that escape javadoc or the embedding comment.
+func formatForeignBlockComment(comment, indent string) string {
+	return formatForeignBlockCommentFrom(comment, 0, indent)
+}
+
+// formatForeignBlockCommentFrom is like [formatForeignBlockComment], except that it
```

Comment:
Style nit for helper functions fHelper that are only called from f and are not supposed to be used anywhere else I personally forgo the doc comments

Reply:
It's not a helper function. It's called directly from the rest of the module. It's used for generating comments like:
```go
fprintf(w, formatForeignBlockCommentFrom("@return "+invoke.Comment, 1))
```

That way any `@` in the `invoke.Comment` string is escaped but the first `@` of the `@return` is left alone.